javascript/ql/src/semmle/javascript/Arrays.qll

```diff
@@ -251,13 +251,17 @@ private module ArrayDataFlow {
   /**
    * A step for creating an array and storing the elements in the array.
    */
-  private class ArrayCreationStep extends DataFlow::AdditionalFlowStep, DataFlow::Node {
-    ArrayCreationStep() { this instanceof DataFlow::ArrayCreationNode }
-
+  private class ArrayCreationStep extends DataFlow::AdditionalFlowStep, DataFlow::ArrayCreationNode {
     override predicate storeStep(DataFlow::Node element, DataFlow::SourceNode obj, string prop) {
       prop = arrayElement() and
-      element = this.(DataFlow::ArrayCreationNode).getAnElement() and
+      element = this.getAnElement() and
       obj = this
+      or
+      exists(int i |
+        element = this.getElement(i) and
+        obj = this and
+        prop = i.toString()
```

Comment:
I think we should have the following instead of the ad-hoc conversion with `.toString`, and also in the name of  hypothetical performance.

```ql
string arrayElement(int i) { i < 5 and result = i.toString() or result = arrayElement() }
```